### PR TITLE
Added shouldIgnoreFileURL:error: convenience wrapper

### PR DESCRIPTION
**Pull request by onecrayon**
Added to GTRepository+Status for compatibility with Swift; closes issue #541.

### CODE REVIEW

**Comment by pietbrauer** [comment]
For my feeling this is a lot of bool comparison to go right the first time. Could you add a test for the new method?

File: ObjectiveGit/GTRepository+Status.m [hunk 1]
```diff
@@ -125,4 +125,10 @@ - (BOOL)shouldFileBeIgnored:(NSURL *)fileURL success:(BOOL *)success error:(NSEr
 	return (ignoreState == 1 ? YES : NO);
 }
 
+- (GTFileIgnoreState)shouldIgnoreFileURL:(NSURL *)fileURL error:(NSError **)error {
+	BOOL success = false;
```
Comment (joshaber) [inline ObjectiveGit/GTRepository+Status.m]: Conventionally this should be `NO` instead of `false`.

File: ObjectiveGit/GTRepository+Status.m [hunk 2]
```diff
@@ -125,4 +125,10 @@ - (BOOL)shouldFileBeIgnored:(NSURL *)fileURL success:(BOOL *)success error:(NSEr
...
+	BOOL ignore = [self shouldFileBeIgnored:fileURL success:&success error:error];
+	return ignore && success ? GTFileIgnoreStateShouldIgnore : (!ignore && success ? GTFileIgnoreStateShouldNotIgnore : GTFileIgnoreStateIgnoreCheckFailed);
```
Comment (joshaber) [inline ObjectiveGit/GTRepository+Status.m]: This is a bit complex to squeeze into one ternary. How about splitting them up?

**Comment by joshaber** [comment]
Thanks @onecrayon! This is close, just a couple comments.

>Could you add a test for the new method?

Yup, a test or two would be great :+1:

File: ObjectiveGit/GTRepository+Status.m (hunk 1 above)
Comment (onecrayon) [inline ObjectiveGit/GTRepository+Status.m]: Ha; guess it shows that I've been coding nothing but Swift for the past while...

**Comment by onecrayon** [comment]
I am not familiar with the testing framework being used here, and there does not appear to be any test associated with the method this is wrapping, so mimicking that is out (e.g. -statusForFile:success:error). I've been testing this locally, and the logic is fine (you'll note my previous pull request fixing the broken logic in the method I'm wrapping...), but if you still want a test added to the suite you'll need to point me in the direction of a good example that I can customize for this particular method.

**Comment by pietbrauer** [comment]
True, the existing implementation is not being tested directly. There are some status specs [here](https://github.com/libgit2/objective-git/blob/master/ObjectiveGitTests/GTRepository%2BStatusSpec.m#L96) you could just copy paste one if them and check if your method returns the right enum value.

**Comment by onecrayon** [comment]
Added a couple of tests: one for the original method and one for the wrapper; both are passing for me. Although FYI the testing suite as a whole is crashing and burning on a memory error for an unrelated test later on:

    -[GTSignatureSpec should_keep_the_git_signature_alive_even_if_the_object_goes_out_of_scope]

**Comment by joshaber** [comment]
Looks good, thanks!

**Pull request merged**